diff --git a/alphafold/alphafold/model/data.py b/alphafold/alphafold/model/data.py
index 249cdb3..55472bf 100644
--- a/alphafold/alphafold/model/data.py
+++ b/alphafold/alphafold/model/data.py
@@ -16,18 +16,12 @@
 
 import io
 import os
-from typing import List
 from alphafold.model import utils
 import haiku as hk
 import numpy as np
 # Internal import (7716).
 
 
-def casp_model_names(data_dir: str) -> List[str]:
-  params = os.listdir(os.path.join(data_dir, 'params'))
-  return [os.path.splitext(filename)[0] for filename in params]
-
-
 def get_model_haiku_params(model_name: str, data_dir: str) -> hk.Params:
   """Get the Haiku parameters from a model name."""
 
